### ISSUE TITLE
Ivy: Should not combine @Input decorators and query decorators

### ISSUE THREAD

**Issue by hugodes** [angular/angular#31327]
# 🐞 bug report

### Affected Package

@angular/compiler-cli (8.1 rc0) 

### Is this a regression?

No, same result without Ivy.

### Description

With Ivy `@Input`decorators shouldn't be combined with query decorators such as `@ViewChild` ( #31056 )

## 🔬 Minimal Reproduction

https://github.com/hugodes/repro-decorators

## 🔥 Exception or Error

The code compiles without errors but should throw an error.

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 8.0.6
Node: 10.16.0
OS: darwin x64
Angular: 8.1.0-rc.0
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.800.6
@angular-devkit/build-angular     0.800.6
@angular-devkit/build-optimizer   0.800.6
@angular-devkit/build-webpack     0.800.6
@angular-devkit/core              8.0.6
@angular-devkit/schematics        8.0.6
@angular/cli                      8.0.6
@ngtools/webpack                  8.0.6
@schematics/angular               8.0.6
@schematics/update                0.800.6
rxjs                              6.5.2
typescript                        3.4.5
webpack                           4.30.0

</code></pre>

**Anything else relevant?**
Sometimes the error is thrown ( #31056 ) but not consistently.

**Comment by istiti** [comment]
I think, this is not an issue, normal behavior.
What you except from angular|cli to throw you without ivy ? I think it should nothing because without ivy it works.

I know it's not something you want to hear but this issue can be closed too because this is how annotation works (in ts you can combine annotation like you do it [here in your code](https://github.com/hugodes/repro-decorators/blob/master/src/app/app.component.ts#L9-L10)) 

If you don't want combine them you need end with "`;`"

To resume 
afaik this alone is not valid `@ViewChild('test', {static: false})` will not create  you class propery `this.test`

You really need write `@ViewChild('test', {static: false}) test;`

cc @AndrewKushnir

**Comment by hugodes** [comment]
Hi @istiti ,

Just to make sure we're on the same page : #31056 referenced the fact that Input decorators and query combinators cannot be combined.
Just to remind you the error was `ERROR in src/app/banky/half-doughnut-chart/half-doughnut-chart.component.ts(10,3): error TS-991006: Cannot combine @Input decorators with query decorators
`
This I'm fine with and thanks to your help, I fixed the error in the code and everything worked well.

Here I'm combining input and query decorators on purpose to show that the compiler doesn't throw an error every-time as I think it should.

In short: combining query and input decorators should fail everytime, not just in some specific cases.

**Comment by istiti** [comment]
Ok, I updated my comment and I see your problem please enable `aot:true`

To understand the problem in JIT (`aot:false` like you have) no angular compiler is used to proceed your ts files; `tsc` is used directly, CLI cannot emit you error in console. 

But when you set `aot:true` angular compiler is used and CLI can throw error you see at runtime.

I admit in little project we can see problem quickly but in large project runtime errors are dangerous! But in v8.1 aot is enabled by default with IVY

cc @alan-agius4

**Comment by hugodes** [comment]
@istiti you're 100% right 🎯. We had aot set to true on our internal project. Never thought to update to aot::true on my repro-repo !

Thank you so much for your time !
